internal/connector/internal/handlers/connectors.go

```diff
@@ -272,6 +267,11 @@ func (h ConnectorsHandler) Patch(w http.ResponseWriter, r *http.Request) {
 				}
 			}
 
+			// If we didn't change anything, then just skip the update...
+			if reflect.DeepEqual(originalResource, resource) {
```

Comment:
I moved the `DeepEqual` to after the validations had run so that I could validate changes to immutable fields with the same approach as the other validations; otherwise I had to treat it slightly different (i.e. either not as a `handlers.Validate` function or at least perform the new check before the `DeepEqual` comparison was made).